contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8677,12 +8688,60 @@ handleGeospatialFunctionsInFunctionCall(TSqlParser::Function_callContext *ctx)
 
 		/* This if-elseIf clause rewrites the query in case of geospatial function calls */
 		if (ctx->spatial_proc_name_server_database_schema()->geospatial_func_arg() && ctx->function_arg_list())
-			rewrite_function_call_geospatial_func_ref_args(ctx);
+		{
+			size_t col_stop_index = ctx->spatial_proc_name_server_database_schema()->column->stop->getStopIndex();
+			size_t func_start_index = ctx->spatial_proc_name_server_database_schema()->geospatial_func_arg()->start->getStartIndex();
+			size_t arg_list_start_index = ctx->function_arg_list()->start->getStartIndex();
+			size_t arg_list_stop_index = ctx->function_arg_list()->stop->getStopIndex();
+
+			rewrite_function_call_dot_func_ref_args(ctx, col_stop_index, func_start_index, arg_list_start_index, arg_list_stop_index);
+		}
 		else if (ctx->spatial_proc_name_server_database_schema()->geospatial_func_no_arg() && !ctx->function_arg_list())
 			rewrite_function_call_geospatial_func_ref_no_arg(ctx);
 	}
 }
 
+static void
+validateXMLFunctionArgs(TSqlParser::Xml_func_argContext *xml_func, TSqlParser::Expression_listContext *expr_list)
+{
+	/* XML EXIST function requires only 1 argument */
+	if (xml_func->EXIST() && (expr_list == NULL || expr_list->expression().size() != 1))
+		throw PGErrorWrapperException(ERROR, ERRCODE_UNDEFINED_FUNCTION, "The exist function requires 1 argument(s).", getLineAndPos(xml_func));
+
+	/* Only String Literal is allowed as agument for XML Functions */
+	if (expr_list)
+	{
+		std::vector<TSqlParser::ExpressionContext *> expression_list = expr_list->expression();
+		for (size_t i = 0; i < expression_list.size(); ++i)
+		{
+			TSqlParser::ExpressionContext *expr = expression_list[i];
+			if (!(dynamic_cast<TSqlParser::Constant_exprContext*>(expr) && static_cast<TSqlParser::Constant_exprContext*>(expr)->constant()->char_string()))
+				throw PGErrorWrapperException(ERROR, 
+						ERRCODE_INVALID_PARAMETER_VALUE, 
+						format_errmsg("The argument %d of the XML data type method \"%s\" must be a string literal.", 
+										(i+1), ::getFullText(xml_func).c_str()), 
+						getLineAndPos(expr));
```

Comment:
Have we done sufficient testing around this to verify? Also have we verified the error message with TSQL?

Reply:
Yes, the tests we had added looks sufficient to test this. Also, verified the error message with TSQL.